gensim/models/word2vec.py

```diff
@@ -1186,12 +1224,60 @@ def scan_vocab(self, sentences, progress_per=10000, trim_rule=None):
                 utils.prune_vocab(vocab, min_reduce, trim_rule=trim_rule)
                 min_reduce += 1
 
+        corpus_count = sentence_no + 1
+        self.raw_vocab = vocab
+        return total_words, corpus_count
+
+    def _scan_vocab_multistream(self, input_streams, workers, trim_rule):
+        manager = multiprocessing.Manager()
+        progress_queue = manager.Queue()
+
+        logger.info("Scanning vocab in %i processes.", min(workers, len(input_streams)))
+
+        workers = min(workers, input_streams)
+        pool = multiprocessing.Pool(processes=workers)
+
+        worker_max_vocab_size = self.max_vocab_size // workers if self.max_vocab_size else None
+        results = [
+            pool.apply_async(_scan_vocab_worker,
+                             (stream, progress_queue, worker_max_vocab_size, trim_rule)
+                             ) for stream in input_streams
+        ]
+        pool.close()
+
+        unfinished_tasks = len(results)
+        total_words = 0
+        total_sentences = 0
+        while unfinished_tasks > 0:
+            report = progress_queue.get()
+            if report is None:
+                unfinished_tasks -= 1
+                logger.info("scan vocab task finished, processed %i sentences and %i words;"
+                            " awaiting finish of %i more tasks", total_sentences, total_words, unfinished_tasks)
+            elif isinstance(report, string_types):
+                logger.warning(report)
+            else:
+                num_words, num_sentences = report
+                total_words += num_words
+                total_sentences += num_sentences
+
+        self.raw_vocab = reduce(utils.merge_counts, [res.get() for res in results])
```

Comment:
Does `.get()` create another copy of each (potentially large) dictionary, or only a reference to an already existing object?

Reply:
Only returns the reference, as far as I know